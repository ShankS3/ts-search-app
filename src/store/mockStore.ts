import thunk from 'redux-thunk';
import { AppDispatch } from 'store/store';
import configureMockStore from 'redux-mock-store';
import { ApplicationState } from 'constants/types';

const middlewares = [thunk];
export const mockStore = configureMockStore<ApplicationState, AppDispatch>(middlewares);