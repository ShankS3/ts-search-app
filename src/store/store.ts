import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER 
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import rootReducer from 'reducers';

const persistConfig = {
  key: "ts-search",
  version: 1,
  storage
}

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER]
      }
    })
});

export let persistor = persistStore(store);

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof rootReducer>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
