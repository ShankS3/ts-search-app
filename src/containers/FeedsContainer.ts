import { FeedState } from 'constants/types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as actions from 'actions/feedActions';
import Feeds from 'components/Feeds';
import { AppDispatch, RootState } from 'store/store';

const mapStateToProps = (state: RootState): { feeds: FeedState } => {
    return {
        feeds: state.feed
    };
};

const mapDispatchToProps = (dispatch: AppDispatch) => {
    return {
        actions: bindActionCreators(actions, dispatch)
    };
};

const FeedsContainer = connect(mapStateToProps, mapDispatchToProps)(Feeds);

export default FeedsContainer;
