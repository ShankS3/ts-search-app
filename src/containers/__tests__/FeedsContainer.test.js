import { mockStore } from 'store/mockStore';
import { render } from '@testing-library/react';
import FeedsContainer from "containers/FeedsContainer";

describe("Feeds Container", () => {

    const initialState = {
        feed: {
            page: 1,
            pageSize: 1,
            data: [],
            totalCount: 0,
            searchText: '',
            sortBy: '',
            loader: false,
            error: false,
            errorMessage: ''
        }
    };

    const feedState = mockStore(initialState);

    it("should match Snapshot", () => {
        const { baseElement } = render(<FeedsContainer store={feedState} />);
        expect(baseElement).toMatchSnapshot();
    })
});
