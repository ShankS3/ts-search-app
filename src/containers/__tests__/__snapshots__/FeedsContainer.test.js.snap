// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feeds Container should match Snapshot 1`] = `
<body>
  <div>
    <div
      class="App"
      data-testid="App"
    >
      <h3
        class="page-title"
        data-testid="page-title"
      >
        Feed
      </h3>
      <div
        class="formbar"
      >
        <div
          class="input-container"
        >
          <div>
            <form>
              <input
                data-testid="search-text"
                id="search-text"
                placeholder="Search..."
                type="text"
              />
              <button
                data-testid="search-button"
                id="search-button"
              >
                <span
                  class="fa fa-search fa-lg"
                />
              </button>
            </form>
          </div>
          <div>
            <label
              for="sort-select"
            >
              Sort by 
            </label>
            <select
              data-testid="sort-select"
              id="sort-select"
            >
              <option>
                sort1
              </option>
              <option>
                sort2
              </option>
              <option>
                sort3
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        class="pagination-wrapper"
        data-testid="pagination-wrapper"
      >
        <button
          class="previous-button"
          data-testid="previous-button"
          disabled=""
        >
           &lt; 
        </button>
        <button
          class="page-button"
          data-testid="page-button"
          disabled=""
        >
           1 
        </button>
        <button
          class="next-button"
          data-testid="next-button"
          disabled=""
        >
           &gt; 
        </button>
      </div>
      <div
        class="card-container"
        data-testid="card-container"
      />
      <div
        class="dataTable-container"
        data-testid="dataTable-container"
      >
        <table
          class="dataTable"
          data-testid="dataTable"
        >
          <thead
            class="dataTable-head"
            data-testid="dataTable-head"
          >
            <tr>
              <th>
                Name
              </th>
              <th>
                Image
              </th>
              <th>
                Description
              </th>
              <th>
                Date Last Edited
              </th>
            </tr>
          </thead>
        </table>
      </div>
    </div>
  </div>
</body>
`;
